### DISCUSSION TITLE
Welcome to Edlib Discussions!

### DISCUSSION THREAD

**Discussion by maxmoeschinger** [cerpus/Edlib#7] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by totoromaum** [comment]
Hi, I am wondering if Edlib has a plan for rewarding systems for content authors. For example, when authors creates a new content, they can have some points. In addition, in order to ensuring the quality of content, content can given "likes." Based on the points and likes of authors, they can be awarded some badges. Thank you for the wonderful work of Edlib team.

**Reply by tmkarlsen** [reply]
Hi @totoromaum 
Especially functionality that can indicate quality, such as usage, reviews or comments, is something we are considering. Interviews with teachers show that honest feedback like tips and suggestions on how to use a resource is the most valuable in determining what a good resource is.
Step two of this could potentially, like you suggest, be connecting these reviews to the user's profile and award badges and XP of some kind, but that is probably a bit ahead. 

We'll look into when we can prioritize adding a review system to our roadmap for development.

We are glad to hear that you like the work we have put down so far and assure you - there is plenty of good stuff to come!

**Reply by totoromaum** [reply]
@tmkarlsen, thank you for the reply. I like the idea of the peer-review system. When contents are good, they tend to have many positive comments, which I think will ensure the quality of the contents. I am delighted to hear that "there is plenty of good stuff to come!" Looking forward to helping teachers create, manage and share content with Edlib.

**Reply by tmkarlsen** [reply]
We've added views in Edlib, @totoromaum. Have a look: https://github.com/cerpus/Edlib/discussions/779

**Reply by totoromaum** [reply]
Nice feature, @tmkarlsen! This will give some rewards to creators.

**Comment by totoromaum** [comment]
Hi, whenever I see the features of Edlib, I find a lot of possibilities for organizations(including my organization) which cannot resort to h5p.com because of various reasons. My organization is using Moodle LMS to support teachers and students. Could you share some roadmaps for Edlib integration to Moodle or other LMSs? In addition, could you inform me of the level of difficulty of integrating Edlib to Moodle LMS? I am asking these because my organization finds strong necessity to manage H5P contents and help teachers share h5p content they created. Thank you for your support. Have a great day.

**Reply by tmkarlsen** [reply]
Hi @totoromaum 
Edlib is made for this scenario - integrating to an application or LMS, thus supporting teachers and students of an organization. Either through a self-hosted setup or a separate setup for the hosted by us. We will be working towards making Edlib easier to integrate to LMSs like Moodle using LTI, and it will probably be part of our goals to complete within the second quarter of 2022. Please refer to https://docs.edlib.com/docs/product/roadmap to see the current roadmap. The Q1 roadmap will be available within this week. 

Once Edlib is set up, integrating towards Moodle should be pretty straightforward. We will add some documentation on this to complete this "Q2" task.

**Reply by totoromaum** [reply]
@tmkarlsen, thank you for the reply. I appreciate it. Looking forward to integrating Edlib to my organization Moodle LMS. Have a great one!

**Reply by tmkarlsen** [reply]
Hi @totoromaum 
Check this out: https://github.com/cerpus/Edlib/discussions/633

**Reply by totoromaum** [reply]
The translation feature is nice and easy. The world users will be happy for this.

**Reply by totoromaum** [reply]
Hi, I am wondering if Edlib LTI features are ready for the integration to LMSs such as Moodle. Please let me know if there are any progresses in this regard. Thank you.

**Reply by tmkarlsen** [reply]
> The translation feature is nice and easy. The world users will be happy for this.

Yes, we really hope so. Note: the last part of Edlib is now available for translation and we'd love your contribution on this part as well: https://weblate.edlib.com/languages/ko/




> Hi, I am wondering if Edlib LTI features are ready for the integration to LMSs such as Moodle. Please let me know if there are any progresses in this regard. Thank you.

We haven't been able to work as dedicated as we wished on this part, but some integration via LTI should be in place. Let us get back to you on this matter over the weekend!

**Comment by totoromaum** [comment]
Hi, I would like to install Edlib on premise(i.e., a high school's server). I am wondering if this is a possible option. Thank you for the wonderful and promising project like Edlib.

**Reply by tmkarlsen** [reply]
Hi, @totoromaum 
We've added a specific thread for [setting up on premise](https://github.com/cerpus/Edlib/discussions/1745)
Follow the steps and report back in that thread and we'll take it from there :)

**Reply by totoromaum** [reply]
Thank you for the documentation, @tmkarlsen. I appreciate it. Have a great one!

**Comment by totoromaum** [comment]
Hi, I hear that NDLA is using Edlib for their own system. What I would like to inquire is if it is possible to use Edlib when a school does not have its own system such as NDLA's and LMSs. The purpose of using Edlib in the school is to support their teachers' creating and sharing H5P content by installing Edlib on premise. Please let me know if Edlib can serve the purpose. Thank you for your support. Have a nice one.

**Reply by tmkarlsen** [reply]
NDLA has their own separate installation of Edlib hosted by us in addition to their custom build CMS.
If I understand you correctly, you ask if Edlib can run as a standalone without any front-end application. 
The answer is no. As of now, Edlib needs a front-end application to launch from. The resources, however, can display standalone.  

So you'll need a front-end application like i.e. Drupal or Moodle. These applications will also enable you to create learning paths/packets containing H5P resources in succession.

**Reply by totoromaum** [reply]
@tmkarlsen, thank you very much for the reply. Could you give a little more explanation about Edlib as resource display standalone? It seems that from what I see based on [this picture of Edlib](https://github.com/cerpus/Edlib/blob/master/sourcecode/docs/docs.edlib.com/static/img/edlib-content-explorer.png), users of Edlib can create and share resources. I understand that frontend applications such as Moodle and Drupal are capable of launching specific h5p contents and grading of h5p activity results, but Edlib standalone is independent of front-end applications if users only use to manage H5P contents, right? Thank you. Have a great one.

**Reply by tmkarlsen** [reply]
Once installed, Edlib can serve the resources to the connected frontend **and** completely standalone like this: https://www.edlib.com/s/resources/bf7135ee-8ffe-43f9-af81-6d15f3c9d412 (we call this standalone view "folium" or "f4m".)

Depending on how you choose to set up Edlib and your frontend-application, you could either use Edlib as an internal application for your organization's members access only (like NDLA uses it) or as an open platform where both internal and external users can create and share resources (like Gamilab.com uses it).  
Edlib is made to be a pure content hub for the creation and serving of interactive content and the belonging data. The frontend application takes care of the contextualization (putting Edlib content together in learning paths), user handling, storing- and displaying of results for the end users (students).

I hope this clarifies what Edlib does and what it doesn't :)

**Reply by totoromaum** [reply]
@tmkarlsen, thank you for the detailed explanation; this clarifies the usages and applications of Edlib. Upon reading your reply, I see the necessity of integrating Edlib to the Moodle LMS. Will the LTI connection processes of Edlib and Moodle be an easy process or a complicated one needing customization? The school which I help will use Moodle 4.0, and Edlib will be integrated to it. Please let me know what kinds of knowledge and skills I need to have to combine the two systems together. Thank you for your time and support. Have a nice one.

**Reply by tmkarlsen** [reply]
At this point, all I can say is that we really can't promise you that it will be an easy road. You'll need developer skills and some knowledge of LTI, and probably more input from our side when the time comes. Unfortunately, the developer who worked on this has left us, and we haven't gotten any progression on the Moodle integration documented.
 
I would recommend that you set up Edlib first and verify that you can access it using Storybook. From there, you'll be able to make sure that all the features of Edlib are up and running, like CRUD of resources, before start integrating with Moodle.

**Reply by tmkarlsen** [reply]
@totoromaum one of our developers have started working on enabling and documenting Moodle integration: https://github.com/cerpus/Edlib/issues/1759

**Reply by totoromaum** [reply]
Hi, @tmkarlsen, this is a great news! Thank you for letting me know. Have a great one!

**Comment by tmkarlsen** [comment]
Hi @totoromaum!
Edlib is now a whole lot easier to set up, and we encourage you to give iHi @totoromaum! Setting up Edlib is now much easier. We encourage you to give it a try. Connecting it to Moodle can also be easily done through LTI.t a try. Connecting it to Moodle is also easily done through LTI.
